### PR TITLE
[CALCITE-5021] Double JOIN is created for NOT IN when IN-list that the values all non-nullable is converted to Values

### PR DESCRIPTION
**Pull request by NobiGo**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java [hunk 1]
```diff
@@ -294,6 +294,33 @@ public static boolean containsNullable(RelDataType type) {
     return false;
   }
 
+  /**
+   * Recreates a RelDataType with the type keep same as the param sourceRelDataType and
+   * nullability keep same as the param targetRelDataType.
```
Comment (asolimando) [inline core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java:299]: What about rephrasing the comment as follows:
```suggestion
   * Creates a RelDataType having the same type of the sourceRelDataType,
   * and the same nullability as the targetRelDataType.
```

File: core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java [hunk 2]
```diff
@@ -294,6 +294,33 @@ public static boolean containsNullable(RelDataType type) {
...
+   */
+  public static RelDataType keepNullableSame(RelDataType sourceRelDataType,
```
Comment (asolimando) [inline core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java:301]: Picking a good name for this method seems hard, what about `keepSourceTypeAndTargetNullability`?

Comment (NobiGo) [reply core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java:301]: Looks good, I will fix it.

File: core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java (hunk 1 above)
Comment (NobiGo) [reply core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java:299]: Looks good, I will fix it.

**Review by asolimando — APPROVED** [review]
LGTM, left few minor comments on javadoc and method naming

**Comment by asolimando** [comment]
@NobiGo please make sure to have a green CI run before merging, the travis failure does not seem related but I'd rather see it running.

**Pull request merged**